dubs/dubs.py

```diff
@@ -0,0 +1,280 @@
+                else:
+                    alignProtLigandDict[pdbID].append(chemID)
+
+            elif SMLigFlag == 1:
+                pdbID = line.split(" ")[0].strip()
+                chemID = line.split(" ")[1].strip()
+
+                if referenceDict.get(curRefPdbID,0) == 0:
+                    referenceDict[curRefPdbID] = [pdbID]
+                else:
+                    referenceDict[curRefPdbID].append(pdbID)
+                
+                if pdbIDSMDict.get(pdbID,0) == 0:
+                    pdbIDSMDict[pdbID] = [chemID]
+                else:
+                    pdbIDSMDict[pdbID].append(chemID)
+
+                entries.add(pdbID)
+
+            elif nonSMligFlag == 1:
+                pdbID = line.split(" ")[0].strip()
+                residueCode = line.split(" ")[1].split("-")[0].strip()
+                chainID = line.split(" ")[1].split("-")[1].strip()
+                residueID = line.split(" ")[1].split("-")[2].strip()
+
+                if referenceDict.get(curRefPdbID,0) == 0:
+                    referenceDict[curRefPdbID] = [pdbID]
+                else:
+                    referenceDict[curRefPdbID].append(pdbID)
+                
+                if pdbIDNonSMDict.get(pdbID,0) == 0:
+                    pdbIDNonSMDict[pdbID] = [tuple([residueCode,chainID,residueID])]
+                else:
+                    pdbIDNonSMDict[pdbID].append(tuple([residueCode,chainID,residueID]))
+
+                entries.add(pdbID)
+            
+            elif noAlignSMFlag == 1:
+                pdbID = line.split(" ")[0].strip()
+                chemID = line.split(" ")[1].strip()
+
+                if noAlignSMDict.get(pdbID,0) == 0:
+                    noAlignSMDict[pdbID] = [chemID]
+                else:
+                    noAlignSMDict[pdbID].append(chemID)
+
+                entries.add(pdbID)
+
+            elif noAlignNonSMFlag == 1:
+                pdbID = line.split(" ")[0].strip()
+                residueCode = line.split(" ")[1].split("-")[0].strip()
+                chainID = line.split(" ")[1].split("-")[1].strip()
+                residueID = line.split(" ")[1].split("-")[2].strip()
+
+                if noAlignNonSMDict.get(pdbID,0) == 0:
+                    noAlignNonSMDict[pdbID] = [tuple([residueCode,chainID,residueID])]
+                else:
+                    noAlignNonSMDict[pdbID].append(tuple([residueCode,chainID,residueID]))
+
+                entries.add(pdbID)
+
+# Define Lemon workflow class
+class MyWorkflow(lemon.Workflow):
+    def __init__(self):
+        lemon.Workflow.__init__(self)
+        
+        self.reference_structures = {}
+        for key, value in pathDict:
+            self.reference_structures[key] = lemon.open_file(value)
+
+    def worker(self, entry, pdbid):
+        # Define and assign the reference pdbid
+        refpdbid = ""
+        # mode is 0 unassigned, 1 for alignment for protein, 2 for alignment for ligand
```

Comment:
for readability, this could use `mode = "unassigned"` explicitly instead of integrers